include/boost/geometry/algorithms/detail/relate/topology_check.hpp

```diff
@@ -181,11 +181,9 @@ struct topology_check<MultiLinestring, Strategy, multi_linestring_tag>
 
         m_has_interior = false;
 
-        typedef typename boost::range_iterator<MultiLinestring const>::type ls_iterator;
-        for ( ls_iterator it = boost::begin(m_mls) ; it != boost::end(m_mls) ; ++it )
+        for (auto it = boost::begin(m_mls); it != boost::end(m_mls); ++it)
         {
-            typename boost::range_reference<MultiLinestring const>::type
-                ls = *it;
+            auto ls = *it;
```

Comment:
This would cause a linestring to be copied. `auto const&` should probably be used here instead.

Reply:
:+1: well spotted, thanks
:heavy_check_mark: changed to `auto const&` and a few lines later (for points) as well.